drivers/afe/ad4110/ad4110.c

```diff
@@ -5,6 +5,22 @@
 #include <stdlib.h>
 #include "ad4110.h"
 #include "error.h"
+#include "irq.h"
+
+int32_t ad4110_spi_int_data_reg_read(struct ad4110_dev *dev,
+				     uint32_t *reg_data);
+
+void irq_adc_read(struct ad4110_callback_ctx *ctx)
+{
+	struct ad4110_dev *dev = ctx->dev;
+	if(ctx->buffer_size > 0) {
+		ad4110_spi_int_data_reg_read(ctx->dev, ctx->buffer);
+		ctx->buffer_size--;
+		ctx->buffer++;
+		irq_enable(dev->irq_desc, dev->nready_pin);
```

Comment:
missing return check

Reply:
This is the irq handler and its type is void.